cli/commands.c

```diff
@@ -359,10 +359,17 @@ int login(int argc, char **argv) {
     fprintf(stderr, "handshake size is invalid in path %s\n", path);
     goto out;
   }
-  if ((handshake[0] & 0x80) != 0) {
-    fprintf(stderr,
-            "only \"service-key-indicator\" prefix type is supported\n");
-    goto out;
+  if ((handshake[0] & 0x80) == 0) {
+    uint8_t public_key[GLOME_MAX_PUBLIC_KEY_LENGTH] = {0};
+    if (glome_derive_key(private_key, public_key)) {
+      fprintf(stderr, "unable to generate a public key\n");
+      goto out;
+    }
+    uint8_t public_key_msb = public_key[GLOME_MAX_PUBLIC_KEY_LENGTH - 1];
+    if ((handshake[0] & 0x7f) != (public_key_msb & 0x7f)) {
```

Comment:
Nit: it should be sufficient to test `handshake[0] != public_key_msb`.

Reply:
Tnx, removed 0x7f from handshake, but not sure it is possible for the key? How do we know high bit is not set here?

Reply:
X25519 public keys are specified to have their most significant bit cleared (https://www.rfc-editor.org/rfc/rfc7748#section-5 second paragraph). That peculiarity is what allowed @l9i to use that bit for metadata in the first place :)

Reply:
Ok, updated with a comment.